### ISSUE TITLE
`baton-chmod` incorrectly ignores the zone property of access controls 

### ISSUE THREAD

**Issue by colin-nolan** [wtsi-npg/baton#156]
`baton-chmod` incorrectly ignores the zone property of access controls (the zone defaults to that in which the target entity resides). The zone however can be "smuggled in" via the `owner` property, using a string in the form `user_or_group#zone`. 

```
$ jq -n '{collection: "/humgen/home/cn13#Sanger1/test"}' | baton-list --acl
{"collection": "/humgen/home/cn13#Sanger1/test", "access": []}

# Attempt to add `cn13` in `Sanger1` as the collection owner using the `zone` property
$ jq -n '{collection: "/humgen/home/cn13#Sanger1/test", access: [{owner: "cn13", zone: "Sanger1", level: "own"}]}' | baton-chmod
{"access": [{"level": "own", "zone": "Sanger1", "owner": "cn13"}], "collection": "/humgen/home/cn13#Sanger1/test"}

# `cn13` in `humgen` was set as the collection owner instead (i.e. zone defaulted to that in which the collection resides)
$ jq -n '{collection: "/humgen/home/cn13#Sanger1/test"}' | baton-list --acl
{"collection": "/humgen/home/cn13#Sanger1/test", "access": [{"owner": "cn13", "zone": "humgen", "level": "own"}]}

# Try again to add `cn13` in `Sanger1` as the collection owner, this time by smuggling the zone in using `#` in the `owner` property
$ jq -n '{collection: "/humgen/home/cn13#Sanger1/test", access: [{owner: "cn13#Sanger1", level: "own"}]}' | baton-chmod
{"collection": "/humgen/home/cn13#Sanger1/test", "access": [{"level": "own", "owner": "cn13#Sanger1"}]}

# `cn13` in `Sanger1` successfully set as owner
$ jq -n '{collection: "/humgen/home/cn13#Sanger1/test"}' | baton-list --acl
{"access": [{"owner": "cn13", "level": "own", "zone": "humgen"}, {"owner": "cn13", "level": "own", "zone": "Sanger1"}], "collection": "/humgen/home/cn13#Sanger1/test"}
```

**Comment by keithj** [comment]
You're correct. The 'user#zone' syntax that iRODS favours was supported first. The JSON-ified version where you don't need to parse the permission string came later and `baton-chmod` has not been updated to use the newer syntax.

**Comment by colin-nolan** [comment]
Am I correct in thinking that this issue was resolved in f5152272ae05b253861668eadf02424e333a8c29?

**Comment by keithj** [comment]
Yes, this is fixed by #159
